### DISCUSSION TITLE
Example: Build Apache Arrow and PyArrow using Xrepo

### DISCUSSION THREAD

**Discussion by cyfdecyf** [xmake-io/xmake-repo#1106] [Show and tell]
I created a project that shows how to build Apache Arrow and PyArrow using Xrepo.

[xrepo-apache-arrow](https://github.com/cyfdecyf/xrepo-apache-arrow)

Apache Arrow has lots' of thirdparty dependencies when you enable many features. By using Xrepo, it's very easy to build Arrow and PyArrow yourself.

The project may also serve as an advanced use of Xrepo as it uses Lua script to pin package versions and configs.

@waruqi Please suggest for improvement. The `add_requireconfs` usage is a little complicated now.

Happy hacking with Xrepo!

**Comment by waruqi** [comment]
https://github.com/cyfdecyf/xrepo-apache-arrow/blob/main/pkg/arrow.lua

try 

```lua
includes("common.lua")

add_requires("arrow", arrow_configs())
add_requireconfs("**.gflags", gflags_configs({override = true}))
add_requireconfs("**.zlib", zlib_configs({override = true}))
add_requireconfs("**.bzip2", bzip2_configs({override = true}))
add_requireconfs("**.boost", boost_configs({override = true}))
add_requireconfs("**.rapidjson", rapidjson_configs({override = true}))
add_requireconfs("**.re2", re2_configs({override = true}))
add_requireconfs("**.utf8proc", utf8proc_configs({override = true}))
add_requireconfs("**.brotli", brotli_configs({override = true}))
add_requireconfs("**.lz4", lz4_configs({override = true}))
add_requireconfs("**.zstd", zstd_configs({override = true}))
add_requireconfs("**.thrift", thrift_configs({override = true}))
add_requireconfs("**.openssl", openssl_configs({override = true}))
add_requireconfs("**.libevent", libevent_configs({override = true}))
add_requireconfs("**.openssl", openssl_configs({override = true}))
```

or 

```lua

includes("common.lua")

add_requires("arrow", arrow_configs())
for _, name for ipairs({"gflags", "zlib", "bz2", ....}) do
    add_requireconfs("**." .. name, get_xxx_configs(name, {override = true}))
end
```

or 

```lua
add_requires("arrow.**." .. name, get_xxx_configs())
```

**Reply by cyfdecyf** [reply]
Thanks for your suggestion. 

I made the change using the second variant in https://github.com/cyfdecyf/xrepo-apache-arrow/commit/091c3b0b6cce4f6cfd22ca398c767eb113fcdb3e , much cleaner now. Please let me know if we can have have more improvements.
